### PR TITLE
ENT-1385 added remind function

### PR DESCRIPTION
**Pull request by irfanuddinahmad**
Description: This PR provides the frontend functionality for sending reminder emails to learners with coupon offer assignments.

Jira: https://openedx.atlassian.net/browse/ENT-1385

### CODE REVIEW

File: src/components/AssignmentReminderModal/emailTemplate.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const emailTemplate = `This is a reminder email that your learning manager has provided you with a access code to take a course at edX. You have redeemed this code {REDEEMED_OFFER_COUNT} of times out of {TOTAL_OFFER_COUNT} number of available course redemptions.
```
Comment (mushtaqak) [inline src/components/AssignmentReminderModal/emailTemplate.js]: Also ` ... number of available course redemptions.` can be `... number of available coupon redemptions` ???

File: src/data/actions/assignmentReminder.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import {
+  ASSIGNMENT_REMINDER_REQUEST,
```
Comment (adamstankiewicz) [inline src/data/actions/assignmentReminder.js]: nits:
1. Rename file to `codeReminder.js`
2. Rename constants to be `CODE_REMINDER_XX`

File: src/data/actions/assignmentReminder.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  ASSIGNMENT_REMINDER_SUCCESS,
+  ASSIGNMENT_REMINDER_FAILURE,
+} from '../constants/assignmentReminder';
+
+import EcommerceApiService from '../services/EcommerceApiService';
+
+const sendAssignmentReminderRequest = () => ({
```
Comment (adamstankiewicz) [inline src/data/actions/assignmentReminder.js]: nit: rename the action functions to be `sendCodeReminderXX`

File: src/data/constants/assignmentReminder.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+const ASSIGNMENT_REMINDER_REQUEST = 'ASSIGNMENT_REMINDER_REQUEST';
```
Comment (adamstankiewicz) [inline src/data/constants/assignmentReminder.js]: nit: Rename file to `codeReminder.js`

File: src/data/reducers/assignmentReminder.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import {
```
Comment (adamstankiewicz) [inline src/data/reducers/assignmentReminder.js]: nit: Rename file to `codeReminder.js`

File: src/data/services/EcommerceApiService.js [hunk 6]
```diff
@@ -35,6 +35,11 @@ class EcommerceApiService {
     const url = `${EcommerceApiService.ecommerceBaseUrl}/api/v2/enterprise/coupons/${couponId}/assign/`;
     return apiClient.post(url, options, 'json');
   }
+
+  static sendAssignmentReminder(couponId, options) {
```
Comment (adamstankiewicz) [inline src/data/services/EcommerceApiService.js]: nit: Rename to `sendCodeReminder`

File: src/containers/AssignmentReminderModal/index.jsx [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import { connect } from 'react-redux';
+
+import AssignmentReminderModal from '../../components/AssignmentReminderModal';
+
+import sendAssignmentReminder from '../../data/actions/assignmentReminder';
+
+const mapDispatchToProps = dispatch => ({
+  sendAssignmentReminder: (couponId, options) => new Promise((resolve, reject) => {
```
Comment (adamstankiewicz) [inline src/containers/AssignmentReminderModal/index.jsx]: nit: `sendCodeReminder`

File: src/components/AssignmentReminderModal/index.jsx [hunk 8]
```diff
@@ -0,0 +1,249 @@
+import React from 'react';
```
Comment (adamstankiewicz) [inline src/components/AssignmentReminderModal/index.jsx]: nit: Rename this component `CodeReminderModal`

**Pull request merged**